python/amici/ode_export.py

```diff
@@ -1989,6 +1990,56 @@ def _compileCCode(self, verbose=False, compiler=None):
         if verbose:
             print(result.stdout.decode('utf-8'))
 
+    def _generateMCode(self):
+        """Create a Matlab script for compiling code files to a mex file
+
+        Arguments:
+
+        Returns:
+
+        Raises:
+
+        """
+        # creating the code lines for the Matlab compile script
+        lines = []
+
+        # Events are not yet implemented. Once this is done, the variable nz
+        # will have to be replaced by "self.model.nz()"
+        nz = 0
+
+        # Second order code is not yet implemented. Once this is done,
+        # those variables will have to be replaced by
+        # "self.model.<var>true()", or the corresponding "model.self.o2flag"
+        nxtrue_rdata = self.model.nx_rdata()
+        nytrue = self.model.ny()
+        o2flag = 0
+
+        # a preliminary comment
+        lines.append('% This compile script was automatically created from Python SBML import.')
+        lines.append('% If mex compiler is set up within MATLAB, it can be run from MATLAB ')
+        lines.append('% in order to compile a mex-file from the Python generated C++ files.')
+        lines.append('')
+
+        # write the actual compiling code
+        lines.append('''modelName = '{model_name}';'''.format(
+            model_name=self.modelName))
+        lines.append('''modelDir = '{model_output_dir}';'''.format(
+            model_output_dir=os.path.abspath(self.modelPath)))
```

Comment:
Maybe it's better to use the relative path here? Let's assume the resulting matlab file always stays within the model directory. The matlab script should just use the mfilepath.

Reply:
Hmmm...
Relative and absolute paths are reasonable to some degree: Personally, I never use the model folders for anything, but would rather move the compile model file. However, it would be easy for me to compile the model in place and then just move the mex file and the simulate file...
Probably you're right, since portability of the generated model is a very nice feature to have...
I'll change that.